src/engine.cr

```diff
@@ -6,62 +6,18 @@ module Engine
 
   # Generates card summary
   struct Crawler
-    getter game_format : String, scry_id : String, foil_status : String, quantity : String
+    getter game_format : String
+    getter scry_id : String
+    getter foil_status : String
+    getter quantity : String
+    getter legality : String
 
-    @card_name : String = ""
-    @set_name : String = ""
-    @set_code : String = ""
-    @rarity : String = ""
-    @legality : String = ""
+    @card : Database::Cards
 
-    @legality_standard : String = ""
-    @legality_future : String = ""
-    @legality_historic : String = ""
-    @legality_gladiator : String = ""
-    @legality_pioneer : String = ""
-    @legality_modern : String = ""
-    @legality_legacy : String = ""
-    @legality_pauper : String = ""
-    @legality_vintage : String = ""
-    @legality_penny : String = ""
-    @legality_commander : String = ""
-    @legality_brawl : String = ""
-    @legality_historicbrawl : String = ""
-    @legality_paupercommander : String = ""
-    @legality_duel : String = ""
-    @legality_oldschool : String = ""
-    @legality_premodern : String = ""
-
-    def initialize(@game_format, @scry_id, @foil_status, @quantity)
-      card_query
-    end
-
-    # Sets card attributes.
-    def card_query : Nil
+    def initialize(@game_format : String, @scry_id : String, @foil_status : String, @quantity : String)
       DB.open "sqlite3://./frightcrawler.db" do |db|
-        db_card = db.query_one "SELECT * from cards where id = ?", "#{@scry_id}", as: Database::Cards
-        @card_name = db_card.name
-        @set_name = db_card.set_name
-        @set_code = "#{db_card.set_code.upcase.colorize.mode(:underline)}"
-        @rarity = db_card.rarity
-
-        @legality_standard = db_card.legality_standard
-        @legality_future = db_card.legality_future
-        @legality_historic = db_card.legality_historic
-        @legality_gladiator = db_card.legality_gladiator
-        @legality_pioneer = db_card.legality_pioneer
-        @legality_modern = db_card.legality_modern
-        @legality_legacy = db_card.legality_legacy
-        @legality_pauper = db_card.legality_pauper
-        @legality_vintage = db_card.legality_vintage
-        @legality_penny = db_card.legality_penny
-        @legality_commander = db_card.legality_commander
-        @legality_brawl = db_card.legality_brawl
-        @legality_historicbrawl = db_card.legality_historicbrawl
-        @legality_paupercommander = db_card.legality_paupercommander
-        @legality_duel = db_card.legality_duel
-        @legality_oldschool = db_card.legality_oldschool
-        @legality_premodern = db_card.legality_premodern
+        @card = db.query_one "SELECT * from cards where id = ?", "#{@scry_id}", as: Database::Cards
```

Comment:
Might also not be a bad to have like `Repository.find @scry_id` that keeps an in memory cache of already fetched cards. This way if the same card is in a csv more than once, it only makes 1 query.

Reply:
I thought you not a fan of double-query method) I was thinking to just drop it - nicer log, and cheaper.